src/decoder/cuda-decoder-utils.cu

```diff
@@ -0,0 +1,237 @@
+// decoder/cuda-decoder-utils.cc
+
+// Copyright      2018  Zhehuai Chen
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http:// www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "fst/fstlib.h"
+#include "cuda-decoder-utils.h"
+
+namespace kaldi {
+
+void get_free_memory_stat(const char *prefix) {
```

Comment:
Functions should be CamelCase (I'm pretty sure of this.)

Reply:
do you mean all functions in this file or only this one? I notice functions in chain-kernels-ansi.h do not use CamelCase style.

Reply:
I noticed as well that chain-kernels-ansi.h do not use CamelCase for function names. But I am pretty sure the rest of the kaldi project uses CamelCase exclusively, so I still suggest being consistent with that (except within chain-kernels-ansi.h. At the very least, we wouldn't want to change the style there in this PR).

Reply:
I decide to change HOST functions to CamelCase and leave DEVICE functions unchanged.